packages/assets-controllers/src/assetsUtil.ts

```diff
@@ -319,7 +319,7 @@ export function divideIntoBatches<Value>(
 }
 
 /**
- * Constructs an object from processing batches of the given values
+ * Constructs a result from processing batches of the given values
```

Comment:
I allowed this to reduce to something other than an object.  In my case an array was useful.  Saw no reason it couldn't be a non-object, same as a regular javascript reduce.  Added tests.